Sources/StreamChat/Config/StreamRuntimeCheck.swift

```diff
@@ -14,4 +14,23 @@ public enum StreamRuntimeCheck {
     ///
     ///  Enables background mapping of DB models
     public static var _isBackgroundMappingEnabled = false
+
+    /// For *internal use* only
+    ///
+    ///  Established the maximum depth of relationships to fetch when performing a mapping
+    ///
+    ///  Eg.
+    ///  Relationship:    Message --->  QuotedMessage --->    QuotedMessage   ---X---     NIL
+    ///  Relationship:    Channel  --->      Message         --->     QuotedMessage  ---X---     NIL
+    ///  Depth:                     0                         1                                     2                               3
+    public static var _backgroundMappingRelationshipsMaxDepth = 2
```

Comment:
- do we need this public actually?
- out of curiosity - what can our customers mess up by changing this value to a bigger one?

Reply:
Nope, was left public for some testing, but should be internal.

> out of curiosity - what can our customers mess up by changing this value to a bigger one?

If the depth makes us reach a loop, it will never finish fetching relationships